packages/s3-store/index.ts

```diff
@@ -25,6 +28,7 @@ type Options = {
   // but may increase it to not exceed the S3 10K parts limit.
   partSize?: number
   useTags?: boolean
+  maxConcurrentPartUploads?: number | Semaphore
```

Comment:
It's not documented that this can also be a `Semaphore`. Any reason against keeping this simple and only doing a `number`, which we always pass to `Semaphore`?

Reply:
I thought that in case a server runs multiple instances of TusServer they might want to share a single Semaphore (which I think in the future we should also allow passing an S3Client too)

Will document now

Reply:
Is that something Supabase already needs? I'm also okay with keeping it simple for now until the need arrises, then we would add a new option which would also be less odd than abusing `maxConcurrentPartUploads` a bit.

Reply:
We don't need it right now, but probably in the near future.
Happy to revisit this later

Reply:
Reverted this